### PR TITLE
Option to keep recent events when purging

### PR DESCRIPTION
**Pull request by TamasSzigeti**
- adding a date arg
- Bump dev deps, some more upkeep and CI fix

### CODE REVIEW

File: src/Infra/Doctrine/DoctrineOutboxStore.php [hunk 1]
```diff
@@ -70,13 +73,20 @@ public function publish(OutboxRecord $outboxRecord): void
...
          * @var OutboxRecordRepository $repo
          */
         $repo = $this->entityManager->getRepository(OutboxRecord::class);
-        $repo->purgePublishedEvents();
+
+        $before = match (true) {
+            $before instanceof DateTimeInterface => $before,
+            $before instanceof DateInterval => (new DateTime())->sub($before),
+            !$before => new DateTime(),
+        };
```
Comment (balazscsaba2006) [inline src/Infra/Doctrine/DoctrineOutboxStore.php:87]: These require proper testing of the repository.

**Review by balazscsaba2006 — CHANGES_REQUESTED** [review]

**Comment by kdaniel95** [comment]
pinging @TamasSzigeti or @balazscsaba2006 in hopes this is not a dead PR.

What is the state of this PR? Does it only need repo testing or does anything else need to be added?

**Comment by balazscsaba2006** [comment]
> pinging @TamasSzigeti or @balazscsaba2006 in hopes this is not a dead PR.
> 
> What is the state of this PR? Does it only need repo testing or does anything else need to be added?

https://github.com/lingoda/domain-events/pull/10#discussion_r1117643767

**Comment by kdaniel95** [comment]
> > pinging @TamasSzigeti or @balazscsaba2006 in hopes this is not a dead PR.
> > What is the state of this PR? Does it only need repo testing or does anything else need to be added?
> 
> [#10 (comment)](https://github.com/lingoda/domain-events/pull/10#discussion_r1117643767)

So functional test on OutboxRecordRepository or on DoctrineOutboxStore? If so, could you provide me with some guidance how it should be done? 
I imagine we create some OutboxRecord, then purge and see if expected records are deleted. I already could achieve creating the schema but cannot really figure out how to populate it (have several problems w/ json_document type for example). 
Thanks for the help!

**Pull request closed without merging**